templates/main.js

```diff
@@ -76,36 +77,49 @@ $(document).ready(function() {
     var cardsHash = undefined;  // hash of most recently rendered current cards, or undefined
     var votesHash = undefined;  // hash of most recently revealed votes, or undefined
     var lastChatUpdate = 0;  // timestamp of most recently retrieved message
+    var games = {};
 
 
     // Periodic Game List Polling
     var gameListWorker = function worker() {
         $.getJSON('getgames', function(data) {
             var html = [];
+            games = {};
             if (data.length > 0) {
-                html.push('<tr><th>&nbsp;</th><th>Host</th><th>Name</th><th>State</th><th>Players</th><th>Score</th><th>Cards<th><th>&nbsp;</th></tr>');
+                html.push('<tr><th>&nbsp;</th><th>Host</th><th>Name</th><th>State</th><th>Players</th><th>Score</th><th>Cards</th><th>Actions</th></tr>');
             }
             $.each(data, function(i, game) {
-                html.push('<tr class="' + (game.gid == activeGame ? 'activeGame' : 'visibleGame') + '">');
+                games[game.gid] = game;
+                html.push('<tr class="' + (game.gid == activeGame ? 'activeGame' : 'visibleGame') + '"'
+                        + ' data-gid="' + game.gid + '">');
                 html.push('<td class="firstCell">' + activityIcon(game.relLastActive) + '</td>');
-                html.push('<td><input type="hidden" name="gid" value="' + game.gid + '" />'
-                        + smilify(game.host) + '</td>');
+                html.push('<td>' + smilify(game.host) + '</td>');
                 html.push('<td>' + smilify(game.name) + '</td>');
                 html.push('<td>' + observerMessages[game.state] + '</td>');
                 html.push('<td title="' + textToHtml(game.players.join(', ')) + '">'
                         + game.players.length
                         + (game.state == {{ states.BEGIN }} ? ' / ' + game.maxPlayers : '') + '</td>');
                 html.push('<td>' + game.topScore + (game.maxScore ? ' / ' + game.maxScore : '') + '</td>');
-                html.push('<td class="lastCell" title="' + game.deckName + '">'
-                        + game.left + ' / ' + game.size + '</td>');
+                html.push('<td title="' + game.deckName + '">' + game.left + ' / ' + game.size + '</td>');
+                html.push('<td class="lastCell">');
+                if (game.isHost) {
+                    html.push('<button class="hide" title="' + hideGameTitle + '">X</button>');
+                }
+                html.push('</td>');
                 html.push('</tr>');
             });
             $('#gameTable').html(html.join(''));
 
             // Register handlers for elements that have been added dynamically
             $('#gameTable button').button();
-            $('.visibleGame').click(function(e) {
-                gid = $(this).find('input').val();
+
+            $('.hide').click(function(e) {
+                gid = $(this).parent().parent().data('gid');
+                openHide(gid, name);
```

Comment:
It looks like `openHide(gid, name)` is called here but the function is `function openHide(gid)` below? Also is there a way to write this so that the `game.name` is passed in here rather than using ` games` dictionary to lookup `games[gid].name` within the function?

Reply:
Nnice catch, the name stuck around here from a previous iteration.

I've tried safely and neatly attaching the name as an attribute, same as the `gid`, but didn't find a clean solution without a lot of refactoring that is safe. Apparently `"` doesn't get encoded into HTML escape sequence and so the way the rows are constructed now, it wouldn't work.

Do you see a problem of storing the game data on client side too? I can limit it to names only if that's better.

Alternatively, I can refactor the game list table construction to build the elements separately which would allow use of the more robust `data` jQuery call to add data on the row.

Reply:
I've ended up refactoring the game list construction with jQuery, this avoids the need for the caching of game data: https://github.com/arvoelke/Dixit/pull/11